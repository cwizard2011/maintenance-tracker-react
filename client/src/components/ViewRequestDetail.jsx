import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ViewInfo, LabelTag } from './common/ViewInfo';

/**
 * @class ViewRequest
 */
class ViewRequestDetail extends Component {
/**
 * @returns {*} jsx
 */
  render() {
    const {
      requestId, sender, title, email, details, status, date,
    } = this.props;

    if (status === 'pending') {
      return (
        <div className="copy">
          <ViewInfo
            requestId={requestId}
            sender={sender}
            title={title}
            email={email}
            details={details}
            date={date}
          />
          <LabelTag
            pClass="label pending"
            labelName="pause"
            status={status}
          />
        </div>
      );
    } else if (status === 'approved') {
      return (
        <div className="copy">
          <ViewInfo
            requestId={requestId}
            sender={sender}
            title={title}
            email={email}
            details={details}
            date={date}
          />
          <LabelTag
            pClass="label success"
            labelName="done-all"
            status={status}
          />
        </div>
      );
    } else if (status === 'resolved') {
      return (
        <div className="copy">
          <ViewInfo
            requestId={requestId}
            sender={sender}
            title={title}
            email={email}
            details={details}
            date={date}
          />
          <LabelTag
            pClass="label success"
            labelName="build"
            status={status}
          />
        </div>
      );
    }
    return (
      <div className="copy">
        <ViewInfo
          requestId={requestId}
          sender={sender}
          title={title}
          email={email}
          details={details}
          date={date}
        />
        <LabelTag
          pClass="label danger"
          labelName="close"
          status={status}
        />
      </div>
    );
  }
}

ViewRequestDetail.propTypes = {
  requestId: PropTypes.string,
  title: PropTypes.string,
  sender: PropTypes.string,
  email: PropTypes.string,
  details: PropTypes.string,
  date: PropTypes.string,
  status: PropTypes.string
};

export default ViewRequestDetail;
