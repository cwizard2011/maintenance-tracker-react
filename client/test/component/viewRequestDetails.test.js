import React from 'react';
import { shallow } from 'enzyme';
import ViewRequestDetail from '../../src/components/ViewRequestDetail';

const props = {
  requestId: '1',
  sender: 'peter',
  title: 'new title',
  email: 'petti4eva@gmail.com',
  details: 'new details',
  status: 'pending',
  date: 'Oct 30, 2018'
};
describe('Table Row', () => {
  it('should render pending details correctly', () => {
    const wrapper = shallow(<ViewRequestDetail {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('should render approved details correctly', () => {
    const newProps = {
      ...props,
      status: 'approved'
    };
    const wrapper = shallow(<ViewRequestDetail {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('should render resolved details correctly', () => {
    const newProps = {
      ...props,
      status: 'resolved'
    };
    const wrapper = shallow(<ViewRequestDetail {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('should render rejected details correctly', () => {
    const newProps = {
      ...props,
      status: 'rejected'
    };
    const wrapper = shallow(<ViewRequestDetail {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(1);
  });
});
