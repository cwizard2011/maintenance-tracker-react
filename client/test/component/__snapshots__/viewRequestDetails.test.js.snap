// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row should render approved details correctly 1`] = `
<div
  className="copy"
>
  <ViewInfo
    date="Oct 30, 2018"
    details="new details"
    email="petti4eva@gmail.com"
    requestId="1"
    sender="peter"
    title="new title"
  />
  <LabelTag
    labelName="done-all"
    pClass="label success"
    status="approved"
  />
</div>
`;

exports[`Table Row should render pending details correctly 1`] = `
<div
  className="copy"
>
  <ViewInfo
    date="Oct 30, 2018"
    details="new details"
    email="petti4eva@gmail.com"
    requestId="1"
    sender="peter"
    title="new title"
  />
  <LabelTag
    labelName="pause"
    pClass="label pending"
    status="pending"
  />
</div>
`;

exports[`Table Row should render rejected details correctly 1`] = `
<div
  className="copy"
>
  <ViewInfo
    date="Oct 30, 2018"
    details="new details"
    email="petti4eva@gmail.com"
    requestId="1"
    sender="peter"
    title="new title"
  />
  <LabelTag
    labelName="close"
    pClass="label danger"
    status="rejected"
  />
</div>
`;

exports[`Table Row should render resolved details correctly 1`] = `
<div
  className="copy"
>
  <ViewInfo
    date="Oct 30, 2018"
    details="new details"
    email="petti4eva@gmail.com"
    requestId="1"
    sender="peter"
    title="new title"
  />
  <LabelTag
    labelName="build"
    pClass="label success"
    status="resolved"
  />
</div>
`;
